test/ssl_test.c

```diff
@@ -160,25 +160,25 @@ static int check_compression(HANDSHAKE_RESULT *result, SSL_TEST_CTX *test_ctx)
 static int check_npn(HANDSHAKE_RESULT *result, SSL_TEST_CTX *test_ctx)
 {
     int ret = 1;
-    ret &= strings_equal("NPN Negotiated (client vs server)",
-                         result->client_npn_negotiated,
-                         result->server_npn_negotiated);
-    ret &= strings_equal("ExpectedNPNProtocol",
-                         test_ctx->expected_npn_protocol,
-                         result->client_npn_negotiated);
+    ret &= strings_eq(result->client_npn_negotiated,
```

Comment:
Here's an idea for a future PR: collect all errors in a static variable and make each test automatically fail if one of the test assertions fails.

Not sure how ugly the implementation would be with the tooling we have (C90) but worth a thought.